javalib/src/main/scala/java/time/LocalDate.scala

```diff
@@ -0,0 +1,432 @@
+  def withDayOfMonth(dayOfMonth: Int): LocalDate =
+    LocalDate.of(year, month, dayOfMonth)
+
+  def withDayOfYear(dayOfYear: Int): LocalDate =
+    LocalDate.ofYearDay(year, dayOfYear)
+
+  override def plus(amount: TemporalAmount): LocalDate =
+    amount.addTo(this).asInstanceOf[LocalDate]
+
+  override def plus(amount: Long, unit: TemporalUnit): LocalDate = unit match {
+    case DAYS => plusDays(amount)
+
+    case WEEKS => plusWeeks(amount)
+
+    case MONTHS => plusMonths(amount)
+
+    case YEARS => plusYears(amount)
+
+    case DECADES =>
+      val years = MathJDK8Bridge.multiplyExact(amount, 10)
+      plusYears(years)
+
+    case CENTURIES =>
+      val years = MathJDK8Bridge.multiplyExact(amount, 100)
+      plusYears(years)
+
+    case MILLENNIA =>
+      val years = MathJDK8Bridge.multiplyExact(amount, 1000)
+      plusYears(years)
+
+    case ERAS => `with`(ERA, MathJDK8Bridge.addExact(get(ERA), amount))
+
+    case _: ChronoUnit =>
+      throw new UnsupportedTemporalTypeException(s"Unsupported unit: $unit")
+
+    case _ => unit.addTo(this, amount)
+  }
+
+  def plusYears(yearsToAdd: Long): LocalDate = {
+    val year1 = year + yearsToAdd
+    requireDateTime(year1 >= minYear && year1 <= maxYear,
+        s"Invalid value for year: $year1")
+    val leap = iso.isLeapYear(year1)
+    val day1 = if (month == 2 && dayOfMonth == 29 && !leap) 28 else dayOfMonth
+    LocalDate.of(year1.toInt, month, day1)
+  }
+
+  def plusMonths(monthsToAdd: Long): LocalDate = {
+    val month1 = month + MathJDK8Bridge.floorMod(monthsToAdd, 12).toInt
+    val year1 = year + MathJDK8Bridge.floorDiv(monthsToAdd, 12) +
+        (if (month1 > 12) 1 else 0)
+    requireDateTime(year1 >= minYear && year1 <= maxYear,
+      s"Invalid value for year: $year1")
+    val month2 = if (month1 > 12) month1 - 12 else month1
+    val day = dayOfMonth min Month.of(month2).length(iso.isLeapYear(year1))
+    LocalDate.of(year1.toInt, month2, day)
+  }
+
+  def plusWeeks(weeksToAdd: Long): LocalDate =
+    plusDays(MathJDK8Bridge.multiplyExact(weeksToAdd, 7))
+
+  def plusDays(daysToAdd: Long): LocalDate = {
+    val epochDay1 = MathJDK8Bridge.addExact(epochDay, daysToAdd)
+    LocalDate.ofEpochDay(epochDay1)
+  }
+
+  override def minus(amount: TemporalAmount): LocalDate =
+    amount.subtractFrom(this).asInstanceOf[LocalDate]
+
+  override def minus(amount: Long, unit: TemporalUnit): LocalDate =
+    if (amount != Long.MinValue) plus(-amount, unit)
+    else plus(Long.MaxValue, unit).plus(1, unit)
+
+  def minusYears(yearsToSubtract: Long): LocalDate =
+    plusYears(-yearsToSubtract)
+
+  def minusMonths(monthsToSubtract: Long): LocalDate =
+    plusMonths(-monthsToSubtract)
+
+  def minusWeeks(weeksToSubtract: Long): LocalDate =
+    minusDays(MathJDK8Bridge.multiplyExact(weeksToSubtract, 7))
+
+  def minusDays(daysToSubtract: Long): LocalDate = {
+    val epochDay1 = MathJDK8Bridge.subtractExact(epochDay, daysToSubtract)
+    LocalDate.ofEpochDay(epochDay1)
+  }
```

Comment:
Hmm, it seems on the JVM `ofEpochDay(0).minusDays(Long.MinValue)` throws a `DateTimeException`, wheras `ofEpochDay(1).minusDays(Long.MinValue)` throws an `ArithmeticException`. So this should probably be replaced by a call to `plusDays(-daysToSubtract)` if `daysToSubtract != Long.MinValue` and `plusDays(Long.MaxValue).plusDays(1)` otherwise.